File: frontend/test_api.js
const { SignJWT } = require("jose");
const crypto = require("crypto");

async function test() {
  const secret = new TextEncoder().encode("super-secret-internal-key-for-dev");
  const jwt = await new SignJWT({ userId: "test@example.com", email: "test@example.com" })
    .setProtectedHeader({ alg: 'HS256' })
    .sign(secret);
    
  const res1 = await fetch("https://cpflow-api.onrender.com/api/users/onboard", {
    method: "POST",
    headers: { "Authorization": `Bearer ${jwt}`, "Content-Type": "application/json" },
    body: JSON.stringify({ name: "Test User" })
  });
  console.log("POST /onboard:", res1.status, await res1.text());
  
  const res2 = await fetch("https://cpflow-api.onrender.com/api/users/profile", {
    headers: { "Authorization": `Bearer ${jwt}` }
  });
  console.log("GET /profile:", res2.status, await res2.text());
}
test();

File: test_api.py
import jwt
import requests

secret = "super-secret-internal-key-for-dev"
token = jwt.encode({"userId": "test@example.com", "email": "test@example.com"}, secret, algorithm="HS256")

headers = {"Authorization": f"Bearer {token}"}

# Test POST /onboard
res1 = requests.post("https://cpflow-api.onrender.com/api/users/onboard", json={"name": "Test User"}, headers=headers)
print("POST /onboard:", res1.status_code, res1.text)

# Test GET /profile
res2 = requests.get("https://cpflow-api.onrender.com/api/users/profile", headers=headers)
print("GET /profile:", res2.status_code, res2.text)
